core/ExceptionHandler.php

```diff
@@ -72,17 +70,28 @@ public static function dieWithCliError($exception)
      */
     public static function dieWithHtmlErrorPage($exception)
     {
-        if ($exception instanceof HttpCodeException
-            && $exception->getCode() > 0
-        ) {
-            http_response_code($exception->getCode());
-        } elseif ($exception instanceof NotYetInstalledException) {
-            http_response_code(404);
-        } else {
-            http_response_code(500);
+        // Set an appropriate HTTP response code.
+        switch (true) {
+            case ( ($exception instanceof \Piwik\Exception\HttpCodeException || $exception instanceof \Piwik\Exception\NotSupportedBrowserException) && $exception->getCode() > 0):
```

Comment:
Might be discussable if the `HttpCodeException` should be within the `Piwik\Exception` namespace, but for now this would be correct:

```suggestion
            case ( ($exception instanceof \Piwik\Http\HttpCodeException || $exception instanceof \Piwik\Exception\NotSupportedBrowserException) && $exception->getCode() > 0):
```